### DISCUSSION TITLE
Retry sending from other IPs on some failures

### DISCUSSION THREAD

**Discussion by KevinGraham-com** [postalserver/postal#1615] [Feature suggestions]
ZoneMTA will automatically retry sending from a different IP if it gets an error response from the target SMTP server that it determines is a blacklist issue. (List of errors here, should be easily editable by server owners: https://github.com/zone-eu/zone-mta/blob/master/config/bounces.txt)

Blacklisted IPs are shown in a section of ZoneMTA's web interface, and the server will not attempt delivery using that IP for 6 hours, allowing time for auto-timeouts to clear the IP as well as manual intervention if needed. This also makes it easy for a server admin to see all issues with deliverability for that server from a single panel.

Would it be possible for Postal to do the same? This should improve deliverability rate, as some types of failures that are currently marked as hard fails would then be able to be retried from other IPs.

**Comment by willpower232** [comment]
This is a nice idea but if you're on lists then you have other problems which should probably be addressed rather than working around.

Also I don't think most Postal installs would have more than one IP address so it wouldn't necessarily be a widely used feature.

**Comment by KevinGraham-com** [comment]
I agree that you should work to get removed from blacklists if there are issues with the emails that you're sending, but just permanently failing an email when it could be sent from a different IP address seems less than ideal.

It seems that this is something that the worker could do when it picks up an email, is to switch the IP address that it's using.
